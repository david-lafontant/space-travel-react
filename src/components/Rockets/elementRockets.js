import { useDispatch } from 'react-redux';
import Button from 'react-bootstrap/Button';
import { PropTypes } from 'prop-types';
import styles from '../../pages/Rockets.module.css';
import { addRocketReservation, removeRocketReservation } from '../../redux/rockets/rockets';

const elementRockets = (props) => {
  const dispatch = useDispatch();
  const {
    id, name, image, desc, reserved,
  } = props;

  const handleReserveClick = () => {
    dispatch(addRocketReservation(id));
  };

  const handleRemoveReserveClick = () => {
    dispatch(removeRocketReservation(id));
  };

  return (
    <li className={styles.rocketContainer} id={id}>
      <img style={{ width: '350px' }} src={image} alt="rocket" />
      <div className={styles.rocketDescription}>
        <h2>{name}</h2>
        <p>
          {reserved && <span>Reserved</span>}
          { `${desc}` }
        </p>
        {reserved ? (
          <Button type="button" className={styles.removeReservation} onClick={handleRemoveReserveClick}>
            Cancel Reservation
          </Button>
        ) : (
          <Button type="button" variant="danger" className={styles.addReservation} onClick={handleReserveClick}>
            Reserve Rocket
          </Button>
        )}
      </div>
    </li>
  );
};

elementRockets.propTypes = {
  name: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  desc: PropTypes.string.isRequired,
  reserved: PropTypes.bool.isRequired,
};

export default elementRockets;
