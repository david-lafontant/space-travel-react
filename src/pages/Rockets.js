import React from 'react';

const Rockets = () => (
  <div>
    <h1>From Rockets</h1>
  </div>
);

export default Rockets;
