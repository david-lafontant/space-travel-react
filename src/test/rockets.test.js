import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import elementStore from '../redux/configureStore';
import Rockets from '../pages/Rockets';

// Arangment
const rockets = [
  {
    desc: 'The SpaceX Rockets are best module for render',
    key: 'frame',
    id: 'rocky',
    type: 'module',
    name: 'rocky',
    image: 'url',
    reservation: false,
  },
];

describe('Test render the Dragon Component', () => {
  it('Dragons section renders a hardcoded object, passed as a prop', () => {
    // Action
    const { container } = render(
      <Provider store={elementStore}>
        <Rockets key={rockets[0].id} dragon={rockets[0]} />
      </Provider>,
    );

    // Assert
    expect(container).toMatchSnapshot();
  });
});