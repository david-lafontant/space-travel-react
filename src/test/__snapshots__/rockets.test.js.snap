// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test render the Rockets Component Rockets section renders a hardcoded object, passed as a prop 1`] = `
<div>
  <ul
    class="flex"
  />
</div>
`;
